pilot/pkg/networking/util/util.go

```diff
@@ -137,24 +128,48 @@ func ListContains(haystack []string, needle string) bool {
 
 // ConvertAddressToCidr converts from string to CIDR proto
 func ConvertAddressToCidr(addr string) *core.CidrRange {
-	if len(addr) == 0 {
-		return nil
+	cidr, err := AddrStrToCidrRange(addr)
+	if err != nil {
+		log.Errorf("failed to convert address %s to CidrRange: %v", addr, err)
 	}
 
-	cidr := &core.CidrRange{
-		AddressPrefix: addr,
-		PrefixLen: &wrapperspb.UInt32Value{
-			Value: getMaxCidrPrefix(addr),
-		},
+	return cidr
+}
+
+// AddrStrToCidrRange converts from string to CIDR proto
+func AddrStrToCidrRange(addr string) (*core.CidrRange, error) {
+	if len(addr) == 0 {
+		return nil, fmt.Errorf("empty address")
 	}
 
+	var (
+		ipAddr        netip.Addr
+		maxCidrPrefix int
+	)
+
 	if strings.Contains(addr, "/") {
-		parts := strings.Split(addr, "/")
-		cidr.AddressPrefix = parts[0]
-		prefix, _ := strconv.Atoi(parts[1])
-		cidr.PrefixLen.Value = uint32(prefix)
+		ipp, err := netip.ParsePrefix(addr)
+		if err != nil {
+			return nil, err
+		}
+		ipAddr = ipp.Addr()
```

Comment:
I think `IsValid()` should be called before this specification.

Reply:
I think `err != nil` already over the `IsValid()`